disperser/common/blobstore/shared_storage.go

```diff
@@ -90,6 +92,24 @@ func (s *SharedBlobStore) StoreBlob(ctx context.Context, blob *core.Blob, reques
 	metadataKey.BlobHash = blobHash
 	metadataKey.MetadataHash = metadataHash
 
+	refreshedMetadata, err := s.GetBlobMetadata(ctx, metadataKey)
+
+	// the only safe condition:
+	// err is disperser.ErrMetadataNotFound && refreshedMetadata is nil
```

Comment:
It looks to me there are 2 failure cases:
- Failed to get metadata (due to errors other than Not Found): this is an INTERNAL error
- Got the metadata: this is FAILED_PRECONDITION error